18_regexp_lab/01.py

```diff
@@ -1,28 +1,13 @@
 import argparse
 import string
 import re
-'''
-parser = argparse.ArgumentParser(description='Returns a parameter that match a given key')
-parser.add_argument('-p','--path', help='definitions file', required = True)
-parser.add_argument('-k','--key', help='key', required = True)
-kuku = parser.parse_args()
 
+#Finally:
 with open(kuku.path, 'r') as f:
     for line in f:
-        temp = line.split()
-        if temp[0].startswith('#'):
+        prefix = re.match(r'\s*#', line)
+        if prefix is not None:
             continue
-        elif temp[0] == kuku.key:
-            res = re.search("=\s+(\w*)", line)
-            print(res.group(1))
-'''
-#After change of attitude :)
-with open(kuku.path, 'r') as f:
-    for line in f:
-        temp = line.split()
-        if temp[0].startswith('#'):
-            continue
-        m = re.search(r'(\w+)\s*=\s*(\w*\s*\w*)', line)
+        m = re.search(r'(\w+)\s*=\s*(.*)', line)
         if m.group(1) == kuku.key and m is not None:
             print(m.group(1), '==>', m.group(2) )
-

```

Comment:
Much better, but we have lost the argparse part from the code so it is a bit hard to try it.
In any case a few issues are left:

* If there are spaces after the value they will be included in the value
* it would be enough to write `if prefix` for the condition in line 9.
* In `if m.group(1) == kuku.key and m is not None:` the conditions are in the wrong order. It would be much better to write `if m and m.group(1) == kuku.key`